python/lsst/obs/base/formatters/fitsExposure.py

```diff
@@ -516,7 +522,107 @@ class FitsExposureFormatter(FitsMaskedImageFormatter):
     type covariance violation ever becomes a practical problem.
     """
 
+    can_read_from_uri = True
     ReaderClass = ExposureFitsReader
+    _cached_fits: tuple[uuid.UUID | None, MemFileManager | None] = (None, None)
+
+    def read_from_uri(self, uri: ResourcePath, component: str | None = None, expected_size: int = -1) -> Any:
+        # For now only support small non-pixel components. In future
+        # could work with cutouts.
+        if uri.isLocal:
+            # For a local URI allow afw to read it directly.
+            return NotImplemented
+        pixel_components = ("mask", "image", "variance")
+
+        if component in pixel_components:
+            # For pixel access currently this can not be cached in memory
+            # and the performance gains are unclear. Assume local file
+            # read with file caching for now.
+            return NotImplemented
+
+        # With current file layouts the non-pixel extensions account for 1/3
+        # of the file size and it is more efficient to download the entire
+        # file.
+        if {"tract", "patch"} & self._dataset_ref.dataId.mapping.keys():
```

Comment:
```suggestion
        if not self._dataset_ref.dataId.mapping.keys().isdisjoint({"tract", "patch"}):
```
may be a little more efficient because it doesn't create a temporary `set`.

Also, FWIW, this check will reject warps, too, which I think is a good thing despite those having small archive sections, because our main access pattern for those is to read the whole thing in sections, and we want the usual local-caching code path for that.